server/src/main/java/org/opensearch/search/pipeline/TrackingSearchResponseProcessorWrapper.java

```diff
@@ -0,0 +1,113 @@
+ * compatible open source license.
+ */
+
+package org.opensearch.search.pipeline;
+
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.core.action.ActionListener;
+
+import java.util.Arrays;
+
+/**
+ * Wrapper for SearchResponseProcessor to track execution details.
+ *
+ *  @opensearch.internal
+ */
+public class TrackingSearchResponseProcessorWrapper implements SearchResponseProcessor {
+
+    private final SearchResponseProcessor wrappedProcessor;
+
+    /**
+     * Constructor for the wrapper.
+     *
+     * @param wrappedProcessor the actual processor to be wrapped
+     */
+    public TrackingSearchResponseProcessorWrapper(SearchResponseProcessor wrappedProcessor) {
+        if (wrappedProcessor == null) {
+            throw new IllegalArgumentException("Wrapped processor cannot be null.");
+        }
+        this.wrappedProcessor = wrappedProcessor;
+    }
+
+    @Override
+    public String getType() {
+        return wrappedProcessor.getType();
+    }
+
+    @Override
+    public String getTag() {
+        return wrappedProcessor.getTag();
+    }
+
+    @Override
+    public String getDescription() {
+        return wrappedProcessor.getDescription();
+    }
+
+    @Override
+    public boolean isIgnoreFailure() {
+        return wrappedProcessor.isIgnoreFailure();
+    }
+
+    @Override
+    public SearchResponse processResponse(SearchRequest request, SearchResponse response) throws Exception {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public SearchResponse processResponse(SearchRequest request, SearchResponse response, PipelineProcessingContext requestContext)
+        throws Exception {
+        ProcessorExecutionDetail detail = new ProcessorExecutionDetail(getType(), getTag());
+
+        long start = System.nanoTime();
+        try {
+            detail.addInput(Arrays.asList(response.getHits().deepCopy().getHits()));
+            SearchResponse result = wrappedProcessor.processResponse(request, response, requestContext);
+            detail.addOutput(Arrays.asList(result.getHits().deepCopy().getHits()));
+            long took = System.nanoTime() - start;
+            detail.addTook(took);
+            requestContext.addProcessorExecutionDetail(detail);
+            return result;
+        } catch (Exception e) {
+            detail.markProcessorAsFailed(ProcessorExecutionDetail.ProcessorStatus.FAIL, e.getMessage());
+            requestContext.addProcessorExecutionDetail(detail);
+            throw e;
+        }
+    }
+
+    @Override
+    public void processResponseAsync(
+        SearchRequest request,
+        SearchResponse response,
+        PipelineProcessingContext requestContext,
+        ActionListener<SearchResponse> responseListener
+    ) {
+        ProcessorExecutionDetail detail = new ProcessorExecutionDetail(getType(), getTag());
+
+        long start = System.nanoTime();
+        try {
```

Comment:
Similar comment to the request processor above. I don't think you need this `try/catch` block.